library/core/src/marker.rs

```diff
@@ -1006,3 +1006,19 @@ pub trait FnPtr: Copy + Clone {
     #[lang = "fn_ptr_addr"]
     fn addr(self) -> *const ();
 }
+
+/// A marker for linear types.
+///
+/// todo: give a clear definition of linear types.
+#[fundamental]
+#[rustc_unsafe_specialization_marker]
+#[unstable(feature = "default_auto_traits", issue = "none")]
+#[cfg_attr(not(bootstrap), lang = "leak")]
+pub unsafe auto trait Leak {}
+
+/// todo
+#[fundamental]
```

Comment:
I'm not sure what exactly restrictions `#[fundamental]` puts on traits, so the fact that we need to add it here concerns me.